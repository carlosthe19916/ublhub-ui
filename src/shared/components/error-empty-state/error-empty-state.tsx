import React from "react";
import { AxiosError } from "axios";
import {
  EmptyState,
  Title,
  EmptyStateBody,
  EmptyStateVariant,
  EmptyStateIcon,
} from "@patternfly/react-core";
import { ConnectedIcon } from "@patternfly/react-icons";

export interface ErrorEmptyStateProps {
  error: AxiosError;
}

export const ErrorEmptyState: React.FC<ErrorEmptyStateProps> = ({ error }) => {
  return (
    <EmptyState variant={EmptyStateVariant.large}>
      <EmptyStateIcon icon={ConnectedIcon} />
      <Title headingLevel="h4" size="lg">
        {error.code}
      </Title>
      <EmptyStateBody>{error.message}</EmptyStateBody>
    </EmptyState>
  );
};
