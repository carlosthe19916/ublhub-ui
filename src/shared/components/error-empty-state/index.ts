export { ErrorEmptyState } from "./error-empty-state";
