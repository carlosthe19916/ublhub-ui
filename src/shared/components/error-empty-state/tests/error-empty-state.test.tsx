import React from "react";
import { shallow } from "enzyme";
import { ErrorEmptyState } from "../error-empty-state";

describe("ErrorEmptyState", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <ErrorEmptyState
        error={{
          code: "404",
          message: "Could not find resource",
          isAxiosError: true,
          name: "My error",
          config: {} as any,
          toJSON: jest.fn(),
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
