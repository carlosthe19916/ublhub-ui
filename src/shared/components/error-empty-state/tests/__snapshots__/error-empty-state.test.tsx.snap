// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorEmptyState Renders without crashing 1`] = `
<EmptyState
  variant="large"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h4"
    size="lg"
  >
    404
  </Title>
  <EmptyStateBody>
    Could not find resource
  </EmptyStateBody>
</EmptyState>
`;
